src/project.js

```diff
@@ -44,18 +52,38 @@ function project(location) {
       throw new Error(`${value} for ${field} is not valid.`);
     }
 
-    options[field] = Object.keys(target);
+    options[field] = Object.keys(forestType);
 
-    if (value && target[value]) {
-      target = target[value];
+    if (value && forestType[value]) {
+      forestType = forestType[value];
     } else {
       // Location does not provide any more values for conditions.
       break;
     }
   }
-  const newLocation = { ...location, options };
-  if (typeof target === 'string') {
-    newLocation.target = target;
+
+  if (typeof forestType !== 'string') {
+    throw new Error(
+      'primitive datatype is not valid for projected forestType.',
```

Comment:
This error message is still very technical and referring to internal implementation details. Something like "Found no projection for location." seems more appropriate. Also a test to catch this error is still missing.

Reply:
Totally, made something like "Found no projection for selected target height.". The test should be in the test file, the third one.